### PR TITLE
Andrei/agora 2142 delegate retirement

### PR DESCRIPTION
**Pull request by andreitr**
Screenshot 2024-06-04 at 12 35 34 PM
Screenshot 2024-06-04 at 12 37 54 PM

---

## PR-Codex overview
The focus of this PR is to add a `delegates` property to `TenantUI` class, allowing for advanced and retired delegates to be stored and accessed.

### Detailed summary
- Added `delegates` property to `TenantUI` class to store advanced and retired delegates
- Defined `UIDelegates` type for advanced and retired delegate addresses
- Updated `TenantUIParams` interface to include `delegates` property
- Implemented logic in `DelegateActions` component to handle retired delegates based on UI configuration

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by yitongzhang** [comment]
LGTM!!

**Review by yitongzhang — APPROVED** [review]

File: src/components/Delegates/DelegateCard/DelegateActions.tsx [hunk 1]
```diff
@@ -24,10 +25,25 @@ export function DelegateActions({
...
   const discord = delegate?.statement?.discord;
   const warpcast = delegate?.statement?.warpcast;
 
+  const isRetired = ui.delegates?.retired.includes(
+    delegate.address.toLowerCase()
+  );
+
+  if (isRetired) {
+    return (
+      <div className="rounded-lg border border-gray-300 p-3 bg-gray-100 text-xs font-semibold text-gray-700">
```
Comment (yitongzhang) [inline src/components/Delegates/DelegateCard/DelegateActions.tsx:40]: Can this be medium sorry lol

**Review by mcgingras — APPROVED** [review]
Awesome! I like the tenant upgrade

**Comment by yitongzhang** [comment]
Jk sorry added 2 visual nits to make this box align
CleanShot 2024-06-04 at 11 44 06@2x

**Pull request merged**